x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransformNodes.java

```diff
@@ -38,14 +41,23 @@ private TransformNodes() {}
 
             Collection<PersistentTasksCustomMetadata.PersistentTask<?>> tasks = tasksMetadata.findTasks(
                 TransformField.TASK_NAME,
-                t -> transformIdsSet.contains(t.getId()) && t.isAssigned()
+                t -> transformIdsSet.contains(t.getId())
             );
 
             for (PersistentTasksCustomMetadata.PersistentTask<?> task : tasks) {
-                executorNodes.add(task.getExecutorNode());
+                if (task.isAssigned()) {
+                    executorNodes.add(task.getExecutorNode());
+                    assigned.add(task.getId());
+                } else {
+                    waitingForAssignment.add(task.getId());
+                }
             }
         }
 
-        return executorNodes.toArray(new String[0]);
+        Set<String> stopped = transformIds.stream()
+            .filter(id -> (assigned.contains(id) || waitingForAssignment.contains(id) == false))
```

Comment:
Stopped transforms are the ones that don't have a task, the id is neither assigned or unassigned.

```suggestion
            .filter(id -> (assigned.contains(id) == false || waitingForAssignment.contains(id) == false))
```

Reply:
:+1: I applied @przemekwitek suggestion and moved the parentheses. There are also some more tests.